improver_tests/calibration/reliability_calibration/test_ConstructReliabilityCalibrationTables.py

```diff
@@ -388,6 +432,28 @@ def test_table_values(self):
         assert_array_equal(result, self.expected_table)
 
 
+class Test__populate_masked_reliability_bins(Test_Setup):
+
+    """Test the _populate_masked_reliability_bins method."""
+
+    def test_table_values_masked_truth(self):
+        """Test the reliability table returned has the expected values when a
+        masked truth is input."""
+
+        forecast_slice = next(self.forecast_1.slices_over("air_temperature"))
+        truth_slice = next(self.masked_truth_1.slices_over("air_temperature"))
+        result = Plugin(
+            single_value_lower_limit=True, single_value_upper_limit=True
+        )._populate_masked_reliability_bins(forecast_slice.data, truth_slice.data)
+
+        self.assertSequenceEqual(result.shape, self.expected_table_shape)
+        self.assertIsInstance(result, np.ma.MaskedArray)
+        assert_array_equal(result.data, self.expected_table_for_mask)
```

Comment:
I've tried to explain why the data underneath the mask is set to zero above.  The zeroing is so that when you do the bitwise addition and potentially add a masked and unmasked point within the reliability tables, the addition will only include the unmasked value.
```
2 (masked) + 3 (unmasked) => 3 (correct answer)
2 (masked) + 3 (unmasked) => 5 (incorrect answer without zeroing the data underneath the mask)
```